app/controllers/bex_controller.rb

```diff
@@ -2,23 +2,33 @@ class BexController < ApplicationController
   before_action :authenticate_user!
 
   def agenda_jap
-    current_date = params.key?(:date) ? params[:date] : Date.today.next_occurring(:friday)
-
-    @appointments = policy_scope(Appointment).for_a_date(current_date).active
+    @appointment_type = AppointmentType.find_by(name: "Sortie d'audience SAP")
+    @current_date = current_date(@appointment_type, params)
+    @agendas = Agenda.in_organization(current_organization).with_open_slots_for_date(@current_date, @appointment_type)
+    @appointments = policy_scope(Appointment).for_a_date(@current_date).active
                                              .joins(slot: [:appointment_type, :agenda])
-                                             .where(appointment_type: { name: "Sortie d'audience SAP" })
+                                             .where('slots.appointment_type_id = ?', @appointment_type.id)
```

Comment:
Tu n'a pas voulu garder le format "en hash" précédent (mais avec l'id) pour une raison particulière? Faudrait checker le formattage précis mais il me semble que ça marcherait

Reply:
Surtout, je viens de réaliser qu'on pourrait éviter ça en ayant un has_many through de appointment_type vers appointments pour faire `policy_scope(@appointment_type.appointments).for_a_date(@current_date).active`. A checker mais je pense que ça marche

Reply:
(à voir pour chainer le `group` ensuite mais ça doit bien se faire)

Reply:
Je ne vois pas trop la logique, mais `.where(appointment_type: @appointment_type)` ne fonctionne pas.
Ça génere `PG::UndefinedColumn: ERROR:  column appointments.appointment_type does not exist`

Reply:
Oui, normal, il faut garder la référence à slots normalement. Mais le has_many through serait sans doute encore mieux ;)